libretroshare/src/pgp/rscertificate.cc

```diff
@@ -330,52 +408,54 @@ bool RsCertificate::initializeFromString(const std::string& instr,uint32_t& err_
 				certificate_version = buf[0];
 				break;
 			case CERTIFICATE_PTAG_PGP_SECTION:
-				binary_pgp_key = new unsigned char[s];
-				memcpy(binary_pgp_key,buf,s);
-				binary_pgp_key_size = s;
+				crt->binary_pgp_key = new unsigned char[s];
+				memcpy(crt->binary_pgp_key, buf, s);
+				crt->binary_pgp_key_size = s;
 				break;
 			case CERTIFICATE_PTAG_NAME_SECTION:
-				location_name = std::string((char *)buf,s);
+				crt->location_name =
+				        std::string(reinterpret_cast<char*>(buf), s);
```

Comment:
The compiler wont let you do that because it is conversion between unrelated pointers type ;) also because of this C++ canst is safer the C cast, because you have to be explicit when you are doing something potentially risky ;)